File: stats-service/model/dto/EventDTO.js
// model/dto/EventDTO.js
class EventDTO {
  constructor(event) {
    this.id = event._id;
    this.eventType = event.eventType;
    this.timestamp = event.timestamp.toISOString(); // Formato estándar
    this.userId = event.userId;
    this.entityId = event.entityId;
    this.entityType = event.entityType;
    this.metadata = event.metadata;
    this.receivedAt = event.receivedAt;
  }
}

module.exports = EventDTO;

File: stats-service/model/dto/EventDTO.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime

class EventDTO(BaseModel):
    eventType: str
    timestamp: datetime
    userId: Optional[str] = None
    anonymous: Optional[bool] = False
    entityType: Optional[str] = None
    entityId: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None